flink-runtime/src/main/java/org/apache/flink/runtime/io/network/partition/consumer/SingleInputGateFactory.java

```diff
@@ -160,16 +159,14 @@ public SingleInputGate create(
                         configuredNetworkBuffersPerChannel,
                         floatingNetworkBuffersPerGate,
                         igdd.getConsumedPartitionType(),
-                        calculateNumChannels(
```

Comment:
If we change the logic directly, other shuffle modes (except for Hybrid Shuffle) may also be affected.